packages/backend/src/managers/llama-stack/llamaStackManager.ts

```diff
@@ -0,0 +1,251 @@
+import { getFreeRandomPort } from '../../utils/ports';
+
+export const LLAMA_STACK_CONTAINER_LABEL = 'ai-lab-llama-stack-container';
+export const LLAMA_STACK_API_PORT_LABEL = 'ai-lab-llama-stack-api-port';
+
+export class LlamaStackManager {
+  #initialized: boolean;
+  #containerInfo: LlamaStackContainerInfo | undefined;
+
+  constructor(
+    private readonly appUserDirectory: string,
+    private taskRegistry: TaskRegistry,
+    private podmanConnection: PodmanConnection,
+    private containerRegistry: ContainerRegistry,
+    private configurationRegistry: ConfigurationRegistry,
+    private telemetryLogger: TelemetryLogger,
+  ) {
+    this.#initialized = false;
+    this.podmanConnection.onPodmanConnectionEvent(this.watchMachineEvent.bind(this));
+    this.containerRegistry.onStartContainerEvent(this.onStartContainerEvent.bind(this));
+    this.containerRegistry.onStopContainerEvent(this.onStopContainerEvent.bind(this));
+  }
+
+  private async refreshLlamaStackContainer(id?: string): Promise<void> {
+    const containers = await containerEngine.listContainers();
+    const containerInfo = containers
+      .filter(c => !id || c.Id === id)
+      .filter(c => c.State === 'running' && c.Labels && LLAMA_STACK_CONTAINER_LABEL in c.Labels)
+      .at(0);
+    if ((id && containerInfo) || !id) {
+      this.#containerInfo = containerInfo
+        ? {
+            containerId: containerInfo.Id,
+            port: parseInt(containerInfo.Labels[LLAMA_STACK_API_PORT_LABEL]),
+          }
+        : undefined;
+    }
+  }
+
+  private async watchMachineEvent(event: PodmanConnectionEvent): Promise<void> {
+    if ((event.status === 'started' && !this.#containerInfo) || (event.status === 'stopped' && this.#containerInfo)) {
+      await this.refreshLlamaStackContainer();
+    }
+  }
+
+  private async onStartContainerEvent(event: ContainerEvent): Promise<void> {
+    await this.refreshLlamaStackContainer(event.id);
+  }
+
+  private onStopContainerEvent(event: ContainerEvent): void {
+    console.log('event id:', event.id, ' containerId: ', this.#containerInfo?.containerId);
+    if (this.#containerInfo?.containerId === event.id) {
+      this.#containerInfo = undefined;
+      this.taskRegistry.deleteByLabels({ trackingId: LLAMA_STACK_CONTAINER_TRACKINGID });
+    }
+  }
+
+  async getLlamaStackContainer(): Promise<LlamaStackContainerInfo | undefined> {
+    if (!this.#initialized) {
+      const containers = await containerEngine.listContainers();
+      this.#containerInfo = containers
+        .filter(c => c.State === 'running' && c.Labels && LLAMA_STACK_CONTAINER_LABEL in c.Labels)
+        .map(c => ({
+          containerId: c.Id,
+          port: parseInt(c.Labels[LLAMA_STACK_API_PORT_LABEL]),
+        }))
+        .at(0);
+      this.#initialized = true;
+    }
+    return this.#containerInfo;
+  }
+
+  async requestCreateLlamaStackContainer(config: LlamaStackContainerConfiguration): Promise<void> {
```

Comment:
Feel weird to have an hardcoded trackingId (`LLAMA_STACK_CONTAINER_TRACKINGID`) when we usually return Promise<string> for `request*` method

https://github.com/containers/podman-desktop-extension-ai-lab/blob/878854ad6ddcc8d8551404b11fd124f595b81e96/packages/backend/src/managers/inference/inferenceManager.ts#L131

what happen if I call it once, delete the container, call it again (without restarting the extension) the tracking Id would be in multiple tasks

Reply:
That is also done in InstructLab I do,'t recall the reason but tasks are cleared once the container is started and that is probably related to the fact that we have a single container to start